exchanges/okx/okx.go

```diff
@@ -0,0 +1,4441 @@
+}
+
+/***********************************Convert Endpoints | Authenticated s*****************************************/
+
+// GetConvertCurrencies retrieves the currency conversion information.
+func (ok *Okx) GetConvertCurrencies(ctx context.Context) ([]ConvertCurrency, error) {
+	var resp []ConvertCurrency
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getConvertCurrenciesEPL, http.MethodGet, assetConvertCurrencies, nil, &resp, true)
+}
+
+// GetConvertCurrencyPair retrieves the currency conversion response detail given the 'currency from' and 'currency to'
+func (ok *Okx) GetConvertCurrencyPair(ctx context.Context, fromCurrency, toCurrency string) (*ConvertCurrencyPair, error) {
+	params := url.Values{}
+	if fromCurrency == "" {
+		return nil, errors.New("missing reference currency name \"fromCcy\"")
+	}
+	if toCurrency == "" {
+		return nil, errors.New("missing destination currency name \"toCcy\"")
+	}
+	params.Set("fromCcy", fromCurrency)
+	params.Set("toCcy", toCurrency)
+	var resp []ConvertCurrencyPair
+	if err := ok.SendHTTPRequest(ctx, exchange.RestSpot, getConvertCurrencyPairEPL, http.MethodGet, common.EncodeURLValues(convertCurrencyPairsPath, params), nil, &resp, true); err != nil {
+		return nil, err
+	}
+	if len(resp) == 1 {
+		return &resp[0], nil
+	}
+	return nil, errNoValidResponseFromServer
+}
+
+// EstimateQuote retrieves quote estimation detail result given the base and quote currency.
+func (ok *Okx) EstimateQuote(ctx context.Context, arg *EstimateQuoteRequestInput) (*EstimateQuoteResponse, error) {
+	if arg == nil {
+		return nil, errNilArgument
+	}
+	if arg.BaseCurrency == "" {
+		return nil, errors.New("missing base currency")
+	}
+	if arg.QuoteCurrency == "" {
+		return nil, errors.New("missing quote currency")
+	}
+	arg.Side = strings.ToLower(arg.Side)
+	if !(arg.Side == order.Buy.Lower() || arg.Side == order.Sell.Lower()) {
+		return nil, errors.New("missing  order side")
+	}
+	if arg.RFQAmount <= 0 {
+		return nil, errors.New("missing rfq amount")
+	}
+	if arg.RFQSzCurrency == "" {
+		return nil, errors.New("missing rfq currency")
+	}
+	var resp []EstimateQuoteResponse
+	err := ok.SendHTTPRequest(ctx, exchange.RestSpot, estimateQuoteEPL, http.MethodPost, assetEstimateQuote, arg, &resp, true)
+	if err != nil {
+		return nil, err
+	}
+	if len(resp) == 1 {
+		return &resp[0], nil
+	}
+	return nil, errNoValidResponseFromServer
+}
+
+// ConvertTrade converts a base currency to quote currency.
+func (ok *Okx) ConvertTrade(ctx context.Context, arg *ConvertTradeInput) (*ConvertTradeResponse, error) {
+	if arg == nil {
+		return nil, errNilArgument
+	}
+	if arg.BaseCurrency == "" {
+		return nil, errors.New("missing base currency")
+	}
+	if arg.QuoteCurrency == "" {
+		return nil, errors.New("missing quote currency")
+	}
+	arg.Side = strings.ToLower(arg.Side)
+	if !(arg.Side == order.Buy.Lower() ||
+		arg.Side == order.Sell.Lower()) {
```

Comment:
```suggestion
	if arg.Side != order.Buy.Lower() && arg.Side != order.Sell.Lower() {
```
What is the side is something else? Its not missing